solutions/instances/main.tf

```diff
@@ -166,16 +192,26 @@ module "cos" {
   existing_cos_instance_id = var.existing_cos_instance_crn
   access_tags              = var.cos_instance_access_tags
   cos_plan                 = "standard"
+}
+
+module "cos_bucket" {
+  depends_on = [time_sleep.wait_for_authorization_policy]
+  providers = {
+    ibm = ibm.cos
+  }
+  count   = (var.existing_log_archive_cos_bucket_name == null || var.existing_at_cos_target_bucket_name == null) ? 1 : 0 # no need to call COS module if consumer is passing existing COS bucket
```

Comment:
I am just thinking...checking length(bucket_config map) will be easier here as all conditions are checked before during bucket config map creation.